core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java

```diff
@@ -97,6 +104,39 @@ public PostgresqlSqlDialect(Context context) {
         SqlParserPos.ZERO);
   }
 
+  @Override public SqlNode rewriteSingleValueExpr(SqlNode aggCall, RelDataType relDataType) {
+    final SqlNode operand = ((SqlBasicCall) aggCall).operand(0);
+    final SqlLiteral nullLiteral = SqlLiteral.createNull(SqlParserPos.ZERO);
+    final SqlNode unionOperand =
```

Comment:
Out of curiosity, why do we cast null here. What if we use **select null union all select null**.

Reply:
When we use `select null union all select null`, The PostgreSQL will throw "ERROR: CASE types text and bigint(The input type) cannot be matched", So we need to tell it the NULL's type.